src/changes.rs

```diff
@@ -1,6 +1,12 @@
 use crate::id::Id;
 use serde::{Deserialize, Serialize};
-use std::collections::{HashMap, VecDeque};
+#[cfg(feature = "std")]
+use std::collections::{hash_map::Entry, HashMap, VecDeque};
+#[cfg(not(feature = "std"))]
+use {
+    alloc::{collections::VecDeque, vec::Vec},
+    hashbrown::{hash_map::Entry, HashMap},
+};
 
 #[derive(Debug, Clone, Serialize, Deserialize, Default)]
```

Comment:
I removed Serialize, Deserialize, and everything compiled.
I'm not sure we really need it. It's the only place the dependency is used in the crate

Reply:
interesting, let's look at that in another pr